### PR TITLE
Add seedetail

### PR DESCRIPTION
**Pull request by chao2048**
Add feature see detail, it works similarly as remove. It will first require the user to enter `seedetail watched` or `seedetail towatch`, then it will prompt the user to enter the index in the watched/ to-watch list. Afterwards, it will show the movie details to the user.

### CODE REVIEW

File: src/main/java/MovieMate.java [hunk 1]
```diff
@@ -84,7 +86,27 @@ public static void main(String[] args) {
...
+                    // list to-watch list for the user to choose
+                    Ui.showListMessage(toWatchList);
+                    System.out.println("Please enter the movie index that you would like to see the detail of it!");
+                    int seeDetailToWatchIndex = Integer.parseInt(inputCommand(scan));
+                    System.out.println(toWatchList.getMovieDetail(seeDetailToWatchIndex));
+                    Ui.showDetailMessage();
+                    break;
+                default:
+                    break;
+                }
```
Comment (zzs-RedCocoon) [inline src/main/java/MovieMate.java:108]: Maybe we can add a help message here
`default:`
` System.out.println("Please follow the format: seedetail [watched/towatch]");`
` break;`

File: src/main/java/MovieList.java [hunk 2]
```diff
@@ -177,4 +177,21 @@ public Movie getMovie(int index) {
...
     }
+
+    public String getMovieDetail(int index) {
+        int i = index - 1;
+        try {
+            Movie movie = this.movieList.get(i);
+            return "Title: " + movie.getTitle() + "\n" +
+                    "Year: " + movie.getYear() + "\n" +
+                    "Genres: " + movie.getGenresString() + "\n" +
+                    "Runtime Minutes: " + movie.getRunTimeMinutes();
```
Comment (zzs-RedCocoon) [inline src/main/java/MovieList.java:188]: I think you can put the return statement at the end to make a happy path.

File: src/main/java/MovieList.java [hunk 3]
```diff
@@ -177,4 +177,21 @@ public Movie getMovie(int index) {
...
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("I don't think you got the right movie number.");
+            e.getMessage();
+        }
+
+        // Do you want to return null or throw new exception?
+        return null;
```
Comment (zzs-RedCocoon) [inline src/main/java/MovieList.java:195]: The "null" here will be printed if user input a false index.
`100`
`>I don't think you got the right movie number.`
`>null`

**Review by zzs-RedCocoon — COMMENTED** [review]
looks good to me

File: src/main/java/MovieList.java (hunk 2 above)
Comment (chao2048) [reply src/main/java/MovieList.java:188]: noted!

File: src/main/java/MovieMate.java (hunk 1 above)
Comment (chao2048) [reply src/main/java/MovieMate.java:108]: okay

File: src/main/java/MovieList.java (hunk 3 above)
Comment (chao2048) [reply src/main/java/MovieList.java:195]: I think this one is written by zhan hong, need to discuss with him.

**Pull request merged**